jcapi.go

```diff
@@ -186,6 +187,38 @@ func (jc JCAPI) Do(op, url string, data []byte) (interface{}, JCError) {
 	return returnVal, err
 }
 
+func (jc JCAPI) DoBytes(op, url string, data []byte) ([]byte, JCError) {
+
+	fullUrl := jc.UrlBase + url
+
+	client := &http.Client{}
+
+	req, err := http.NewRequest(op, fullUrl, bytes.NewReader(data))
+	if err != nil {
+		return nil, fmt.Errorf("ERROR: Could not build search request: '%s'", err)
+	}
+
+	jc.setHeader(req)
+
+	resp, err := client.Do(req)
+	if err != nil {
+		return nil, fmt.Errorf("ERROR: client.Do() failed, err='%s'", err)
+	}
+
+	defer resp.Body.Close()
+
+	if resp.Status != "200 OK" {
+		return nil, fmt.Errorf("JumpCloud HTTP response status='%s'", resp.Status)
+	}
+
+	buffer, err := ioutil.ReadAll(resp.Body)
+	if err != nil {
+		return nil, fmt.Errorf("ERROR: Could not read the response body, err='%s'", err)
+	}
+
+	return buffer, err
+}
+
```

Comment:
Not sure about feelings on TODO comments, but it'd be nice to note somewhere that `Do` should be avoided in favor of `DoBytes` for any new code until a point in time at which they are collapsed into one function.